### PR TITLE
Embed templates in their generators

### PR DESCRIPTION
**Pull request by whatyouhide**

### CODE REVIEW

**Comment by whatyouhide** [comment]
cc @ericmj @britto

**Review by ericmj — APPROVED** [review]

File: lib/protobuf/protoc/generator/service.ex [hunk 1]
```diff
@@ -20,12 +30,15 @@ defmodule Protobuf.Protoc.Generator.Service do
...
     generate_desc = if ctx.gen_descriptors?, do: desc, else: nil
-    Protobuf.Protoc.Template.service(mod_name, name, methods, generate_desc)
+    service_template(mod_name, name, methods, generate_desc)
   end
 
   defp generate_service_method(ctx, method) do
     input = service_arg(Util.type_from_type_name(ctx, method.input_type), method.client_streaming)
-    output = service_arg(Util.type_from_type_name(ctx, method.output_type), method.server_streaming)
+
+    output =
```
Comment (britto) [inline lib/protobuf/protoc/generator/service.ex:39]: Can't spot the diff here. Maybe a formatter issue?

**Review by britto — APPROVED** [review]
This seems to reduce unnecessary indirection. I'm in 👍.

File: lib/protobuf/protoc/generator/service.ex (hunk 1 above)
Comment (whatyouhide) [reply lib/protobuf/protoc/generator/service.ex:39]: Yep, formatter

**Pull request merged**